src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -451,10 +451,16 @@ public ActivityProfileState isActive( final AlignmentContext context, final Refe
 
         final int ploidy = activeRegionEvaluationGenotyperEngine.getConfiguration().genotypeArgs.samplePloidy;
         final List<Allele> noCall = GATKVariantContextUtils.noCallAlleles(ploidy); // used to noCall all genotypes until the exact model is applied
-        final Map<String, AlignmentContext> splitContexts = context.splitContextBySampleName(readsHeader);
+
+        final Map<String, AlignmentContext> splitContexts;
+        if (sampleSet.size() == 1) {
+            splitContexts = context.splitContextBySampleName(samplesList.getSample(0), readsHeader);
```

Comment:
Also, it's strange that you're checking `sampleSet.size()`, then getting the sample from `samplesList`